source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -32,44 +37,197 @@ ZooKeeperFilterConfig::ZooKeeperFilterConfig(const std::string& stat_prefix,
+  initOpCode(OpCodes::Sync, stats_.sync_resp_, stats_.sync_resp_fast_, stats_.sync_resp_slow_,
+             "sync_resp");
+  initOpCode(OpCodes::Check, stats_.check_resp_, stats_.check_resp_fast_, stats_.check_resp_slow_,
+             "check_resp");
+  initOpCode(OpCodes::Multi, stats_.multi_resp_, stats_.multi_resp_fast_, stats_.multi_resp_slow_,
+             "multi_resp");
+  initOpCode(OpCodes::Reconfig, stats_.reconfig_resp_, stats_.reconfig_resp_fast_,
+             stats_.reconfig_resp_slow_, "reconfig_resp");
+  initOpCode(OpCodes::SetWatches, stats_.setwatches_resp_, stats_.setwatches_resp_fast_,
+             stats_.setwatches_resp_slow_, "setwatches_resp");
+  initOpCode(OpCodes::SetWatches2, stats_.setwatches2_resp_, stats_.setwatches2_resp_fast_,
+             stats_.setwatches2_resp_slow_, "setwatches2_resp");
+  initOpCode(OpCodes::CheckWatches, stats_.checkwatches_resp_, stats_.checkwatches_resp_fast_,
+             stats_.checkwatches_resp_slow_, "checkwatches_resp");
+  initOpCode(OpCodes::RemoveWatches, stats_.removewatches_resp_, stats_.removewatches_resp_fast_,
+             stats_.removewatches_resp_slow_, "removewatches_resp");
+  initOpCode(OpCodes::GetEphemerals, stats_.getephemerals_resp_, stats_.getephemerals_resp_fast_,
+             stats_.getephemerals_resp_slow_, "getephemerals_resp");
   initOpCode(OpCodes::GetAllChildrenNumber, stats_.getallchildrennumber_resp_,
+             stats_.getallchildrennumber_resp_fast_, stats_.getallchildrennumber_resp_slow_,
              "getallchildrennumber_resp");
-  initOpCode(OpCodes::Close, stats_.close_resp_, "close_resp");
+  initOpCode(OpCodes::Close, stats_.close_resp_, stats_.close_resp_fast_, stats_.close_resp_slow_,
+             "close_resp");
 }
 
-void ZooKeeperFilterConfig::initOpCode(OpCodes opcode, Stats::Counter& counter,
-                                       absl::string_view name) {
+void ZooKeeperFilterConfig::initOpCode(OpCodes opcode, Stats::Counter& resp_counter,
+                                       Stats::Counter& resp_fast_counter,
+                                       Stats::Counter& resp_slow_counter, absl::string_view name) {
   OpCodeInfo& opcode_info = op_code_map_[opcode];
-  opcode_info.counter_ = &counter;
+  opcode_info.resp_counter_ = &resp_counter;
+  opcode_info.resp_fast_counter_ = &resp_fast_counter;
+  opcode_info.resp_slow_counter_ = &resp_slow_counter;
   opcode_info.opname_ = std::string(name);
   opcode_info.latency_name_ = stat_name_set_->add(absl::StrCat(name, "_latency"));
 }
 
+absl::flat_hash_map<int32_t, std::chrono::milliseconds>
+ZooKeeperFilterConfig::parseLatencyThresholds(LatencyThresholdList latency_thresholds) {
+  absl::flat_hash_map<int32_t, std::chrono::milliseconds> latency_threshold_map;
+  for (const auto& threshold : latency_thresholds) {
+    switch (threshold.opcode()) {
+    case envoy::extensions::filters::network::zookeeper_proxy::v3::LatencyThreshold::Default:
```

Comment:
let's try some DRY

introducing 
```
using LatencyThreshold_Opcode = envoy::extensions::filters::network::zookeeper_proxy::v3::LatencyThreshold_Opcode;
using LatencyThreshhold = envoy::extensions::filters::network::zookeeper_proxy::v3::LatencyThreshold;

int32 getOpCodeIndex(LatencyThreshold_Opcode code) {
  static constexpr absl::flat_hash_map<LatencyThreshold_Opcode, int32_t> opcode_map = {
  // Use a special key to store the default latency threshold.
    {LatencyThreshold::Default, -999}, 
   {LatencyThreshold::Connect, 0}, ...
  }
 }

  ASSERT(opcode_map.find(code) != opcode_map.end());
  return opcode_map[code];
}
``` 
and here we could do
```
latency_threshold_map[getOpcodeIndex(threshold.opcode()] =
          std::chrono::milliseconds(PROTOBUF_GET_MS_REQUIRED(threshold, threshold));
```

Reply:
Also could we move the check `// Check duplicated opcodes in config.` to here?

Reply:
i,e., remove the `std::set` section and throw exception here if dup

Reply:
> Also could we move the check `// Check duplicated opcodes in config.` to here?

I think the opcode duplication check related to config validation, so I put it in `config.cc`.